app/core/services/campaign_service.py

```
@@ -601,6 +601,113 @@ async def archive_campaign_service(campaign_id: int, db: AsyncSession) -> Campai
+        campaign_id: The campaign to pause
+        db: AsyncSession
+    Returns:
+        CampaignRead: The updated campaign
+    Raises:
+        CampaignNotFoundError: if campaign does not exist
+        InvalidStateTransitionError: if transition is not valid
+    """
+    result = await db.execute(select(Campaign).where(Campaign.id == campaign_id))
+    campaign = result.scalar_one_or_none()
+    if not campaign:
+        raise CampaignNotFoundError(f"Campaign {campaign_id} not found")
```

Ruff (0.15.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)